src/app/item-details/item-details.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { DataService } from '../shared/services/data.service';
+import { Item } from '../shared/models/item.model';
+
+@Component({
+  selector: 'app-item-details',
+  templateUrl: './item-details.component.html',
+  styleUrls: ['./item-details.component.scss'],
+  providers: [DataService]
+})
+export class ItemDetailsComponent implements OnInit, OnDestroy {
+
+  itemId: number;
+  item: Item;
+  sub: any;
+
+  constructor(
+    private route: ActivatedRoute,
+    private dataService: DataService
+  ) { }
+
+  ngOnInit() {
+
+    this.sub = this.route.params.subscribe(params => {
```

Comment:
I looked into ActivatedRoutes and Observables, and based on Angular's API: https://angular.io/guide/router#observable-parammap-and-component-reuse, there's no need to unsubscribe, because when the component is destroyed, Angular automatically unsubscribes to the observable. We can remove the unsubscribe, and the variable `this.sub`.

Also, I don't think it is necessary to add extra memory for the variable, `itemId`.

```
import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { DataService } from '../shared/services/data.service';
import { Item } from '../shared/models/item.model';

@Component({
  selector: 'app-item-details',
  templateUrl: './item-details.component.html',
  styleUrls: ['./item-details.component.scss'],
  providers: [DataService]
})
export class ItemDetailsComponent implements OnInit {
  item: Item;

  constructor(
    private route: ActivatedRoute,
    private dataService: DataService
  ) {}

  ngOnInit() {
    this.route.params.subscribe(params => this.item = this.dataService.getItemById(+params['id']));
  }
}
```